adam-core/src/main/java/org/bdgenomics/adam/io/FastqRecordReader.java

```diff
@@ -105,13 +107,31 @@
     private static final int MAX_LINE_LENGTH = 10000;
 
     /**
+     * True if the underlying data is compressed with a splittable codec.
+     */
+    private boolean compressedAndSplittable = false;
+    
+    /**
+     * True if the last read was <=0 bytes in size.
```

Comment:
Alas, the splittable codecs use -2 to signify end of splittable block and -1 to signify end of stream. I'll fix the `<` though.